columnflow/production/cms/seeds.py

```diff
@@ -18,8 +18,8 @@
 @producer(
     uses={
         # global columns for event seed
-        "run", "luminosityBlock", "event", "nGenJet", "nGenPart", "nJet", "nPhoton", "nMuon",
-        "nElectron", "nTau", "nSV",
+        "run", "luminosityBlock", "event",
+        "Photon.pt", "SV.pt",
```

Comment:
Are you reading the "pt" fields to trigger that the counter fields are read too?
Couldn't one just keep the "n*" fields here? It might be very costly to e.g. read all GenPart pt's just for the sake of knowing how many particles exist per event.

Reply:
Yes, `pt` is just a proxy in this case so we can get the counts via `ak.num`. But I agree that the overhead might not be justified. This was a bit of a last-minute change since I was seeing issues with `n*`, but I will look into what was causing that and revert this if I can find a solution.

On a related note, an advantage of not relying on the `n*` fields would be that `deterministic_seeds` could then be run during `cf.ProduceColumns` and later, where these fields are not available. Is that something we want to support?

Reply:
> On a related note, an advantage of not relying on the n* fields would be that deterministic_seeds could then be run during cf.ProduceColumns and later, where these fields are not available. Is that something we want to support?

Good question. I think for that we would need to make sure that the counter columns (or similar ones) are available. Maybe we can split implementation from the `used` declaration, e.g.

```python
@producer(
    uses={"run", "luminosityBlock", "event"},  # commonly-used columns
    produces={"deterministic_seed"},
)
def deterministic_event_seeds_impl(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
    # implementation
    ...


# specialization using counter columns
@producer(
    uses={
        deterministic_event_seeds_impl,
        "nGenJet", "nGenPart", "nJet", "nPhoton", "nMuon",
        "nElectron", "nTau", "nSV",
        # first-object columns for event seed
        "Tau.jetIdx", "Tau.decayMode",
        "Muon.jetIdx", "Muon.nStations",
        "Jet.nConstituents", "Jet.nElectrons", "Jet.nMuons",
    },
    produces={deterministic_event_seeds_impl},
)
def deterministic_event_seeds(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
    return self[deterministic_event_seeds_impl](events, **kwargs)
```

and then re-use the implementation at a later stage in case we need it.

Reply:
I agree, this is more complicated than I anticipated. Let's keep this for now and put the actual issue at hand on our lists.